apps/aecore/lib/aecore/tx/data_tx.ex

```diff
@@ -255,14 +261,20 @@ defmodule Aecore.Tx.DataTx do
              tx_type_state,
              block_height,
              payload,
-             tx
+             tx,
+             context
            ) do
       new_chainstate =
-        if tx.type.get_chain_state_name() == :accounts do
-          %{chainstate | accounts: new_accounts_state}
-        else
-          %{chainstate | accounts: new_accounts_state}
-          |> Map.put(tx.type.get_chain_state_name(), new_tx_type_state)
+        case tx.type.get_chain_state_name do
+          :accounts ->
+            %{chainstate | accounts: new_accounts_state}
+
+          :contracts ->
```

Comment:
I see now, how you made the return values from `process_chainstate` of contract txs work. So to fix the problem (using `Chain.chain_state` in txs) you should pass `chainstate` as the tx_type_state argument to `process_chainstate`.